main.ts

```diff
@@ -23,29 +24,125 @@ export default class MyPlugin extends Plugin {
+						let cursorPositionInSentence = cursorPosition.ch-sentence.index;
+						let cutPortion=sentence[0].substring(0,cursorPositionInSentence);
+						let newPar=paragraphText.replace(cutPortion," ");
+						editor.setLine(cursorPosition.line,newPar);
+						editor.setCursor({line:cursorPosition.line,ch:cursorPosition.ch-cutPortion.length});
+					}
+				}
+			}
+		});
+
+		this.addCommand({
+			id: 'Delete-forward-sentence',
+			name: 'Delete to end of sentence',
+			hotkeys: [{ modifiers: ["Mod","Shift"], key:"Delete"}],
+			editorCallback: (editor: Editor) => {
+				const wholeSentenceRegex = (/(==(.*?)==)|[^.!?\s][^.!?]*(?:[.!?](?!['"]?\s|$)[^.!?]*)*[.!?]?['"]?(?=\s|$)/gm);
+				const cursorPosition = editor.getCursor();
+				console.log("This is the cursor position", cursorPosition)
+				let paragraphText = editor.getLine(cursorPosition.line);
+				//let sentences = paragraphText.matchAll(/[.?!] /gm);
+				let sentences = paragraphText.matchAll(wholeSentenceRegex);
+				for(const sentence of sentences) {
+					if ( cursorPosition.ch >=sentence.index && cursorPosition.ch <sentence.index+sentence[0].length){
+						let cursorPositionInSentence = cursorPosition.ch-sentence.index;
+						let cutPortion=sentence[0].substring(cursorPositionInSentence);
+						console.log(cutPortion);
+						let newPar=paragraphText.replace(cutPortion,"");
+						editor.setLine(cursorPosition.line,newPar);
+						editor.setCursor({line:cursorPosition.line,ch:cursorPosition.ch});
+						// editor.setCursor(sentence.index);
+					}
+				}
+			}
+		});
+
+		this.addCommand({
+			id: 'Move-backwards-sentence',
+			name: 'Move to beginning of sentence',
+			hotkeys: [{ modifiers: ["Alt"], key:"Left"}],
+			editorCallback: (editor: Editor) => {
+				const wholeSentenceRegex = (/(==(.*?)==)|[^.!?\s][^.!?]*(?:[.!?](?!['"]?\s|$)[^.!?]*)*[.!?]?['"]?(?=\s|$)/gm);
+				let cursorPosition = editor.getCursor();
+				let paragraphText = editor.getLine(cursorPosition.line);
+//				console.log("This is the cursor position", cursorPosition)
+				if (cursorPosition.ch==0){
+					let previousParText=editor.getLine(cursorPosition.line-1);
+//					console.log(previousParText);
+					editor.setCursor({line:cursorPosition.line-1,ch:previousParText.length-1});
+				}
+				else {
+					const sentences = paragraphText.matchAll(wholeSentenceRegex);
+					for(const sentence of sentences) {
+						if ( cursorPosition.ch >=sentence.index && sentence.index+sentence[0].length >= cursorPosition.ch){
+							//console.log(` Cursor is at char ${cursorPosition.ch} on a line of length ${editor.getLine(cursorPosition.line).length}`);
+							editor.setCursor({line:cursorPosition.line,ch:sentence.index-2});
+						}
+					}
+					// editor.setCursor(sentence.index);
+				}
+			}
+		});
+
+		this.addCommand({
+			id: 'Move-forwards-sentence',
+			name: 'Move to start of next sentence',
+			hotkeys: [{ modifiers: ["Mod"], key:"Right"}],

```

Comment:
Probably don't want to bind to shortcuts which override OS defaults (on Mac, `Alt`+`Left` moves to the beginning of the previous word and `Mod`+`Right` moves to the end of the line)

Reply:
That is a good point. Although I use a macbook when I am out and about my fingers can never remember the OS text movement defaults. I get repeatedly frustrated when Cmd-Left/Right does not move by the word, as it does on Windows. I think the answer is to pile on the modifier keys for these movements. What would be the Mac equivalent of the "Win" key?

Reply:
Users can always set their own hotkeys for these in Obsidian settings if they want. That way, they can ensure it doesn't conflict with any part of their workflow.

The Mac equivalent to the Windows key, according to the [official docs](https://support.apple.com/en-au/guide/mac-help/cpmh0152/mac), is actually Command as well (so same as Ctrl).